### PR TITLE
Feat/chats new

### PR DESCRIPTION
**Pull request by nirKoppAC**

### CODE REVIEW

File: src/app/_components/chats/ChatInput.tsx
Comment (Dolevkle) [inline src/app/_components/chats/ChatInput.tsx:1]: I don't understand why you need to the from ref

File: src/app/_components/chats/ChatInput.tsx [hunk 1]
```diff
@@ -56,21 +70,43 @@ export default function ChatInput({ chatId }: Props) {
...
   };
 
+  useEffect(() => {
+    form.register("imageUrl", { required: false });
+  }, [form.register]);
+
+  useEffect(() => {
+    console.log(form.formState.errors);
+  }, [form.formState.errors]);
+
```
Comment (Dolevkle) [inline src/app/_components/chats/ChatInput.tsx:95]: I believe both of them are redundant, if not explain me and we'll think together of a solution without using use effect

File: src/app/_components/chats/ImageUploadDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    <Dialog open={isOpen} onOpenChange={setIsOpen}>
+      <DialogTrigger asChild>
+      {Button}
+      </DialogTrigger>
+      <DialogContent>
+        <ImageUpload
+          onUploadSuccess={(url) => {
+            onUploadSuccess(url); // Call the parent's function
+            setIsOpen(false); // Close the dialog
+          }}
```
Comment (Dolevkle) [inline src/app/_components/chats/ImageUploadDialog.tsx:27]: move it to upper function called handleUploadSuccess and then it'll be in the html
<ImageUpload onUploadSuccess={handleUploadSuccess}/>

File: src/app/_components/home/ProfilePicture/ProfilePictureDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        <ImageUpload
+          onUploadSuccess={() => {
+            onUploadSuccess();
+            toast({
+              title: "Success",
+              description: "Image uploaded successfully",
+            });
+            setIsOpen(false);
+          }}
+        />
```
Comment (Dolevkle) [inline src/app/_components/home/ProfilePicture/ProfilePictureDialog.tsx:38]: same here

File: src/app/_components/home/ProfilePicture/ProfilePictureDisplay.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    ? usersImages.map((image) => image.url)
+    : ["https://utfs.io/f/9dc01f95-3ff0-484f-822e-59500be4baeb-x8m314.png"];
+
+  const handleDeleteSuccess = async () => {
+    await refetchImages();
+  };
+
+  const handleUploadSuccess = async () => {
+    await refetchImages();
+  };
```
Comment (Dolevkle) [inline src/app/_components/home/ProfilePicture/ProfilePictureDisplay.tsx:36]: these can be one liners and they're same function maybe they are redundant

File: src/app/_components/loading-button.tsx
Comment (Dolevkle) [inline src/app/_components/loading-button.tsx:1]: I told elad also I believe this is redundant. we have Button component and we can see how they do loading button n shadcn. you can see further in his cr

File: src/app/_components/chats/ChatInput.tsx
Comment (nirKoppAC) [reply src/app/_components/chats/ChatInput.tsx:1]: It is used to submit the form using the code, that's the best solution I found

File: src/app/_components/chats/ChatInput.tsx (hunk 1 above)
Comment (nirKoppAC) [reply src/app/_components/chats/ChatInput.tsx:95]: I removed them

File: src/app/_components/chats/ImageUploadDialog.tsx (hunk 2 above)
Comment (nirKoppAC) [reply src/app/_components/chats/ImageUploadDialog.tsx:27]: Done

**Pull request merged**